src/client/datascience/notebook/kernelWithMetadata.ts

```diff
@@ -37,8 +42,31 @@ export class VSCodeNotebookKernelMetadata implements VSCNotebookKernel {
         private readonly kernelProvider: IKernelProvider,
         private readonly notebook: IVSCodeNotebook,
         private readonly context: IExtensionContext,
-        private readonly preferredRemoteKernelIdProvider: PreferredRemoteKernelIdProvider
-    ) {}
+        private readonly preferredRemoteKernelIdProvider: PreferredRemoteKernelIdProvider,
+        private readonly notebookKernelProvider: INotebookKernelProvider
+    ) {
+        this.notebookKernelProvider.onDidGetFontAwesomeMessage((e) => {
```

Comment:
I'd prefer we move this code into the `kernelProvider.ts` file.
This way we do'nt need to add `onDidGetFontAwesomeMessage` (fewer API changes).